test/framework/easyconfig.py

```diff
@@ -1132,6 +1133,49 @@ def test_hide_hidden_deps(self):
         self.assertEqual(ec['hiddendependencies'][0]['full_mod_name'], 'toy/.0.0-deps')
         self.assertEqual(ec['dependencies'], [])
 
+    def test_quote_str(self):
+        """Test quote_str function."""
+        self.assertEqual(quote_str('foo'), '"foo"')
+        self.assertEqual(quote_str('foo\'bar'), '"foo\'bar"')
+        self.assertEqual(quote_str('foo\'bar"baz'), '"""foo\'bar"baz"""')
+        self.assertEqual(quote_str("foo'bar\"baz"), '"""foo\'bar"baz"""')
```

Comment:
add test cases with `\n` in the string, both without specifying `escape_newline` (to test the default), and with setting it to `True` and `False`